dali/operators/audio/mfcc/mfcc.cu

```diff
@@ -0,0 +1,96 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/audio/mfcc/mfcc.h"
+#include <vector>
+#include "dali/kernels/signal/dct/dct_gpu.h"
+#include "dali/core/static_switch.h"
+
+namespace dali {
+
+namespace detail {
+
+__global__ void CalcLifterKernel(float *coeffs, int64_t start_idx, int64_t target_length,
+                                 float lifter) {
+  float ampl_mult = lifter / 2;
+  float phase_mult = M_PI / lifter;
+  for (int64_t i = start_idx + threadIdx.x; i < target_length; i += blockDim.x)
+    coeffs[i] = 1.0 + ampl_mult * sinf(phase_mult * (i + 1));
+}
+
+template <>
+DLL_PUBLIC  void LifterCoeffs<GPUBackend>::Calculate(int64_t target_length, float lifter,
+                                                     cudaStream_t stream)  {
+  // If different lifter argument, clear previous coefficients
+  if (lifter_ != lifter) {
+    coeffs_.clear();
+    lifter_ = lifter;
+  }
+
+  // 0 means no liftering
+  if (lifter_ == 0.0f)
+    return;
+
+  // Calculate remaining coefficients (if necessary)
+  if (static_cast<int64_t>(coeffs_.size()) < target_length) {
+    int start_idx = coeffs_.size();
+    int added_length = target_length - start_idx;
+    coeffs_.resize(target_length, stream);
+    int threads = std::min(added_length, 256);
+    CalcLifterKernel<<<1, threads, 0, stream>>>(coeffs_.data(), start_idx, target_length, lifter);
+  }
+}
+
+}  // namespace detail
+
+template<>
+bool MFCC<GPUBackend>::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                 const workspace_t<GPUBackend> &ws) {
+  GetArguments(ws);
+  auto &input = ws.InputRef<GPUBackend>(0);
+  TYPE_SWITCH(input.type().id(), type2id, T, MFCC_SUPPORTED_TYPES, (
+    using Kernel = kernels::signal::dct::Dct1DGpu<T>;
+    kmgr_.Initialize<Kernel>();
+    kmgr_.Resize<Kernel>(1, 1);
+    ctx_.gpu.stream = ws.stream();
+    output_desc.resize(1);
+    output_desc[0].type = input.type();
+    auto in_view = view<const T>(input);
+    auto &req = kmgr_.Setup<Kernel>(0, ctx_, in_view, make_cspan(args_), axis_, span<float>());
+    output_desc[0].shape = req.output_shapes[0];
+    int64_t max_ndct = 0;
+    for (int i = 0; i < nsamples_; ++i) {
+      int64_t ndct = output_desc[0].shape[i][axis_];
+      if (ndct > max_ndct) max_ndct = ndct;
+    }
+    lifter_coeffs_.Calculate(max_ndct, lifter_, ws.stream());
+  ), DALI_FAIL(make_string("Unsupported data type: ", input.type().id())));  // NOLINT
+  return true;
+}
+
+template<>
+void MFCC<GPUBackend>::RunImpl(workspace_t<GPUBackend> &ws) {
+  auto &input = ws.InputRef<GPUBackend>(0);
+  TYPE_SWITCH(input.type().id(), type2id, T, MFCC_SUPPORTED_TYPES, (
+    using Kernel = kernels::signal::dct::Dct1DGpu<T>;
+    auto in_view = view<const T>(input);
+    auto out_view = view<T>(ws.OutputRef<GPUBackend>(0));
+    span<const float> lifter_span(lifter_coeffs_.data(), lifter_coeffs_.size());
```

Comment:
As mentioned before, for GPU data use a tensor view:
```suggestion
    auto lifter_coeffs = make_tensor_gpu<1>(lifter_coeffs_.data(), {lifter_coeffs_.size()});
```

Reply:
done